packages/editor/src/components/post-url/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { safeDecodeURIComponent, cleanForSlug } from '@wordpress/url';
+import { useState } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { TextControl, ExternalLink } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+export default function PostURL( { onClose } ) {
+	const { isEditable, postSlug, postLink, permalinkPrefix, permalinkSuffix } =
+		useSelect( ( select ) => {
+			const permalinkParts = select( editorStore ).getPermalinkParts();
+			return {
+				isEditable: select( editorStore ).isPermalinkEditable(),
+				postSlug: safeDecodeURIComponent(
+					select( editorStore ).getEditedPostSlug()
+				),
+				postLink: select( editorStore ).getCurrentPost().link,
+				permalinkPrefix: permalinkParts?.prefix,
+				permalinkSuffix: permalinkParts?.suffix,
+			};
+		}, [] );
+
+	const { editPost } = useDispatch( editorStore );
+
+	const [ forceEmptyField, setForceEmptyField ] = useState( false );
+
+	return (
+		<div className="editor-post-url">
+			<InspectorPopoverHeader title={ __( 'URL' ) } onClose={ onClose } />
+			{ isEditable && (
+				<TextControl
+					label={ __( 'Permalink' ) }
+					value={ forceEmptyField ? '' : postSlug }
+					autoComplete="off"
+					spellCheck="false"
+					help={
+						<>
+							{ __( 'The last part of the URL.' ) }{ ' ' }
+							<ExternalLink
+								href={ __(
+									'https://wordpress.org/support/article/settings-sidebar/#permalink'
+								) }
+							>
+								{ __( 'Learn more.' ) }
+							</ExternalLink>
+						</>
+					}
+					onChange={ ( newValue ) => {
+						editPost( { slug: newValue } );
+						// When we delete the field the permalink gets
+						// reverted to the original value.
+						// The forceEmptyField logic allows the user to have
+						// the field temporarily empty while typing.
+						if ( ! newValue ) {
+							if ( ! forceEmptyField ) {
+								setForceEmptyField( true );
+							}
+							return;
+						}
+						if ( forceEmptyField ) {
+							setForceEmptyField( false );
+						}
+					} }
+					onBlur={ ( event ) => {
+						editPost( {
+							slug: cleanForSlug( event.target.value ),
+						} );
+						if ( forceEmptyField ) {
+							setForceEmptyField( false );
+						}
+					} }
+				/>
+			) }
+			<h3 className="editor-post-url__link-label">{ __( 'URL' ) }</h3>
```

Comment:
I've no strong preference here, but using "View Post" (`postType.labels.view_item`) seemed more user-friendly. cc @javierarce

Reply:
Good point, yes, let's do this. It also avoid the redundancy of having two labels with the same name.